modelforge/train/parameters.py

```diff
@@ -0,0 +1,313 @@
+"""
+This will define a pydantic model for the training parameters, meant to take the dictionary defined in the
+associated yaml file. This model will be used to validate the input dictionary and to provide default values.
+
+
+
+"""
+
+from pydantic import BaseModel, model_validator, field_validator, ConfigDict
+from enum import Enum
+from typing import List, Union, Dict, Optional, Callable
+
+
+# So we  do not need to set Config parameters in each model
+# we can create a base class that will be inherited by all models
+class ParametersBase(BaseModel):
+    model_config = ConfigDict(
+        use_enum_values=True, arbitrary_types_allowed=True, validate_assignment=True
+    )
+
+
+class SchedulerMode(str, Enum):
+    """
+    Enum class for the scheduler mode, allowing "min" ond "max"
+
+    args:
+        min (str): The minimum mode
+        max (str): The maximum mode
+    """
+
+    min = "min"
+    max = "max"
+
+
+class ThresholdMode(str, Enum):
+    """
+    Enum class for the threshold mode, allowing "abs" and "rel"
+
+    args:
+        abs (str): The absolute mode
+        rel (str): The relative mode
+    """
+
+    abs = "abs"
+    rel = "rel"
+
+
+# generate an enum class for each registered splitting strategy
+# NOTE: not sure if it is best to allow this to dynamically update
+# or to simply hardcode in the values.  Hardcoded values would be better for
+# documentation, which is really the aim of adding in the dataclasses
+# from modelforge.dataset.utils import REGISTERED_SPLITTING_STRATEGIES
+#
+# SplittingStrategyName = Enum(
+#     "SplittingStrategyName", {name: name for name in REGISTERED_SPLITTING_STRATEGIES}
+# )
+
+
+class SplittingStrategyName(str, Enum):
+    """
+    Enum class for the splitting strategy name
+
+    args:
+        first_come_first_serve (str): The first come first serve strategy
+        random_record_splitting_strategy (str): Split based on records randomly (i.e., all conformers of a molecule are in the same set)
+        random_conformer_splitting_strategy (str): Split based on conformers randomly (i.e., conformers of a molecule can be in different sets)
+    """
+
+    first_come_first_serve = "first_come_first_serve"
+    random_record_splitting_strategy = "random_record_splitting_strategy"
+    random_conformer_splitting_strategy = "random_conformer_splitting_strategy"
+
+
+class AnnealingStrategy(str, Enum):
+    """
+    Enum class for the annealing strategy
+    """
+
+    cos = "cos"
+    linear = "linear"
+
+
+class Loggers(str, Enum):
+    """
+    Enum class for the experiment logger
+    """
+
+    wandb = "wandb"
```

Comment:
wandb needs a few additional parameters, how can we best add these?
```
        logger = WandbLogger(
            save_dir=save_dir,
            log_model=True,
            project=wandb_config["project_name"],
            group=wandb_config["group"],
            name=experiment_name,
            job_type=wandb_config["job_type"],
        )
```

Reply:
Since we only have two logger options at the moment, we can create a tensorboard class (with relevant parameters) and a wandb class.    I'm going to create and issue for this so we can do it in a separate, targeted PR.

Reply:
I created issue #209 so we don't lose this comment